HOW_TO.md

```diff
@@ -103,22 +107,113 @@ any template .yaml files.
-12. Open Kibana url and verify workload index populate with data:
-   1. Create the workload index: Kibana -> Hamburger tab -> Stack Management -> Index patterns -> Create index pattern -> workload-results -> timestamp -> Done
-   2. Verify workload-results index is populated: Kibana -> Hamburger tab -> Discover -> workload-results (index) -> verify that there is a new data
+6. Create dedicate `workload class` StressngPod or StressngVM in dedicate module `stressng_pod.py` or `stressng_vm.py` and customized workload run steps accordingly [stressng_pod.py](benchmark_runner/benchmark_operator/stressng_pod.py) 
+7. Add workload method name (workload_pod/workload_vm) to environment_variables_dict['workloads'] in [environment_variables.py](benchmark_runner/main/environment_variables.py)
+8. Create workload folder in the [templates](benchmark_runner/common/template_operations/templates) directory.  Create the following files in that directory:
+   1. Add workload_data_template for configuration parameters, e.g. [stressng_data_template.yaml](benchmark_runner/common/template_operations/templates/stressng/stressng_data_template.yaml).  
+   2. The data template is structured as discussed [below](#data-template).
+   3. Add workload pod and VM custom resource template inside [internal_data](benchmark_runner/common/template_operations/templates/stressng/internal_data)
+9. Add workload folder path in [MANIFEST.in](MANIFEST.in), add 2 paths: the workload path to 'workload_data_template.yaml' and path to 'internal_data' Pod and VM template yaml files
+   ```
+     include benchmark_runner/common/template_operations/templates/stressng/*.yaml
+     include benchmark_runner/common/template_operations/templates/stressng/internal_data/*.yaml
+   ```
+10. Add tests for all new methods you write under `tests/integration`.
+11. Update the golden unit test files as described [above](#add-new-workload-modify-parameters-to-workload-or-change-parameters-for-any-ci-job)
+12. For test and debug workload, need to configure [environment_variables.py](benchmark_runner/main/environment_variables.py)
+13. Fill parameters: workload, kubeadmin_password, pin_node_benchmark_operator, pin_node1, pin_node2, elasticsearch, elasticsearch_port
+14. Run [main.py](/benchmark_runner/main/main.py)  and verify that the workload run correctly
+15. The workload can be monitored and checked through 'current run' folder inside the run workload flavor (default flavor: 'test_ci')
+16. Open Kibana url and verify workload index populate with data:
+17. Create the workload index: Kibana -> Hamburger tab -> Stack Management -> Index patterns -> Create index pattern -> workload-results -> timestamp -> Done
+18. Verify workload-results index is populated: Kibana -> Hamburger tab -> Discover -> workload-results (index) -> verify that there is a new data
+
+## Add new custom workload to benchmark runner
```

Comment:
This really is a lot of steps to run to add a new workload; we need to look at simplifying the procedure.

Reply:
Yes you write, I guess that we will automate it soon.

Reply:
Agreed; we're not going to solve every problem at once.